src/dstack/_internal/core/backends/cloudrift/api_client.py

```diff
@@ -0,0 +1,228 @@
+        if isinstance(response_data, dict):
+            return response_data.get("instance_ids", [])
+        return []
+
+    def list_instances(self, instance_ids: Optional[List[str]] = None) -> List[Dict]:
+        request_data = {
+            "selector": {
+                "ByStatus": ["Initializing", "Active", "Deactivating"],
+            }
+        }
+        logger.debug("Listing instances with request data: %s", request_data)
+        response_data = self._make_request("instances/list", request_data)
+        if isinstance(response_data, dict):
+            return response_data.get("instances", [])
+
+        return []
+
+    def get_instance_by_id(self, instance_id: str) -> Optional[Dict]:
+        request_data = {"selector": {"ById": [instance_id]}}
+        logger.debug("Getting instance with request data: %s", request_data)
+        response_data = self._make_request("instances/list", request_data)
+        if isinstance(response_data, dict):
+            instances = response_data.get("instances", [])
+            if isinstance(instances, list) and len(instances) > 0:
+                return instances[0]
+
+        return None
+
+    def is_instance_ready(self, instance_id: str) -> bool:
+        """
+        Checks if the instance with the given ID is ready.
+        Returns True if the instance is ready, False otherwise.
+        """
+        instance_info = self.get_instance_by_id(instance_id)
+        if instance_info:
+            instance_type = instance_info.get("node_mode", "")
+            if instance_type == "VirtualMachine":
+                vms = instance_info.get("virtual_machines", [])
+                if len(vms) > 0:
+                    vm_ready = vms[0].get("ready", False)
+                    return vm_ready
+            else:
+                return instance_info.get("status", "") == "Active"
+        return False
+
+    def terminate_instance(self, instance_id: str) -> bool:
+        request_data = {"selector": {"ById": [instance_id]}}
+        logger.debug("Terminating instance with request data: %s", request_data)
+        response_data = self._make_request("instances/terminate", request_data)
+        if isinstance(response_data, dict):
+            info = response_data.get("terminated", [])
+            return len(info) > 0
+
+        return False
+
+    def _make_request(
+        self,
+        endpoint: str,
+        data: Optional[Mapping[str, Any]] = None,
+        method: str = "POST",
+        **kwargs,
+    ) -> Union[Mapping[str, Any], str, Response]:
+        headers = {}
+        if self.api_key is not None:
+            headers["X-API-Key"] = self.api_key
+
+        version = CLOUDRIFT_API_VERSION
+        full_url = f"{self.public_api_root}/{endpoint}"
+
+        try:
+            response = requests.request(
+                method,
+                full_url,
+                headers=headers,
+                json={"version": version, "data": data},
+                timeout=120,
```

Comment:
Blocking for that long can overload the `dstack` server. If possible, I suggest setting a much lower timeout.
```suggestion
                timeout=15,
```